File: Data Structures & Algorithms/baseball-game/submission-1.java
class Solution {
    public int calPoints(String[] operations) {
        Deque<Integer> st = new ArrayDeque<>();

        for (String s : operations) {
            if (s.equals("+")) {
                int last = st.pop();
                int second = st.peek();

                st.push(last);
                st.push(last + second);

            } else if (s.equals("D")) {
                st.push(st.peek() * 2);

            } else if (s.equals("C")) {
                st.pop();

            } else {
                st.push(Integer.parseInt(s));
            }
        }

        int sum = 0;
        for (int val : st) sum += val;

        return sum;
    }
}

File: Data Structures & Algorithms/time-needed-to-buy-tickets/submission-1.py
class Solution:
    def timeRequiredToBuy(self, tickets: List[int], k: int) -> int:
        q=deque(tickets)
        t=0
        while q[k]>1 :
            #print(q , k)
            p=q.popleft()
            p-=1
            if p>=1 :

                q.append(p)

            k-=1
            if k<0 :
                k=len(q)-1
            t+=1
        return t+len(q)
        """
        5111 0
        1114 3
        114 2
        14 1
        4 0
        3 0
        2 0
        1 
        """
        